src/weather_gen.cpp

```diff
@@ -26,81 +27,97 @@ int weather_generator::current_winddir = 1000;
 w_point weather_generator::get_weather( const tripoint &location, const time_point &t,
                                         unsigned seed ) const
 {
-    const double x( location.x /
-                    2000.0 ); // Integer x position / widening factor of the Perlin function.
-    const double y( location.y /
-                    2000.0 ); // Integer y position / widening factor of the Perlin function.
-    const double z( to_turn<int>( t + calendar::season_length() ) /
-                    2000.0 ); // Integer turn / widening factor of the Perlin function.
+    static const time_duration season_length = calendar::season_length();
+    static const time_duration year_length = calendar::year_length();
+    static constexpr int weather_cache_limit = 10000000;
+    static const w_point weather_cache_value_not_found_in_cache = { 0.0, 0.0, 0.0, 0.0, "NFIC", 0, false };
+    static lru_cache<weather_cache_key_type, w_point> weather_cache;
```

Comment:
I feel like all these `static const` values should be members of `weather_generator`, not function-static.  Otherwise it could cause issues when saving one game and loading another in a different world.
(The `constexpr` one is fine).

Reply:
I am thinking of making calendar values members of `game` class, so they would only be read from options only once on game start.

Reply:
I wouldn't do that. What we want is decoupled classes with separated concerns without compromising code quality for performance reasons. Instead, you could decouple `calendar` from `options` by introducing setters for the options it reads. Think of it in terms of testing: as for now, `calendar` cannot be tested without mocking `options`. Ideally, classes should be self sustainable and capable of being tested in isolation. Such are highly reusable and easy-to-test classes.

Reply:
Thanks for the tip. I will remove time-related changes from here and will make changes to `calendar` class in another pull request.

Reply:
I think it makes sense to do both: the calendar values should be members of a `calendar` class, but that in turn should be accessed via `game`, not be a global as it currently is.

Reply:
I skimmed through the code and game inits a single calendar on game start. I guess I will manage setting values read from options there and remove options dependency from calendar class at all.